text/0000-try-expr.md

```diff
@@ -0,0 +1,886 @@
+
+6. Degree / Risk of breakage.
+
+7. Consistency with old learning material.
+
+    1. Inversely: The extent of the old learning material
+
+   That is, (in)consistency with `?` and the `try!()` macro.
+   If the first clause is called `try`,
+   then `try { }` and `try!()` would have essentially inverse meanings.
+
+## Rationale for `try`
+[rationale for try]: #rationale-for-try
+
+1. **Fidelity to the construct's actual behavior:** Very high
+2. **Precedent from existing languages:** A lot, see [prior-art]
+    1. **Popularity of the languages:** Massive accumulated dominance
+    2. **Fidelity to behavior in those languages:** Very high
+    3. **Familiarity with respect to their analogous constructs:** Very high
+3. **Brevity / Length:** 3
+4. **Consistency with the naming of the trait used for `?`:** Consistent
+5. **Consistency with related libstd fn conventions:** Consistent
+6. **Risk of breakage:** High (if we assume `try!` will break, otherwise: Low)
+    - **Used in std:** [*No*](https://doc.rust-lang.org/nightly/std/?search=try), `std::try!`, but it is technically possible to not break this macro. (unstable: `std::intrinsics::try` so irrelevant)
+    - **Used as crate?** [*Yes*](https://crates.io/crates/try). No reverse dependencies. Described as: *"Deprecation warning resistant try macro"*
+    - **Usage (sourcegraph):** **27** regex:
+    ```
+    repogroup:crates case:yes max:400
+    \b((let|const|type|)\s+try\s+=|(fn|impl|mod|struct|enum|union|trait)\s+try)\b
+    ```
+7. **Consistency with old learning material:** Inconsistent ([`try!`])
+
+### Review
+
+This is our choice of keyword, because it:
+
+1. has a massive dominance in both popular and less known languages and is
+   sufficiently semantically faithful to what `try` means in those languages.
+2. is consistent with the standard library wrt. `Try` and `try_` prefixed methods.
+3. it is brief.
+4. it has high fidelity wrt. the concepts it attempts to communicate (exception boundary for `?`).
+5. it can be further extended with `catch { .. }` handlers if we wish.
+
+## Alternative: reserving `catch`
+
+1. **Fidelity to the construct's actual behavior:** Low
+2. **Precedent from existing languages:** None
+3. **Brevity / Length:** 6
+4. **Consistency with the naming of the trait used for `?`:** Inconsistent
+5. **Consistency with related libstd fn conventions:** Somewhat consistent
+6. **Risk of breakage:** Low
+    - **Used in std:** [*No*](https://doc.rust-lang.org/nightly/std/?search=catch)
+    - **Used as crate?** [*No*](https://crates.io/crates/catch).
+    - **Usage (sourcegraph):** **21** regex:
+    ```
+    repogroup:crates case:yes max:400
+    \b((let|const|type|)\s+catch\s+=|(fn|impl|mod|struct|enum|union|trait)\s+catch)\b
+    ```
+7. **Consistency with old learning material:** Untaught
+
+### Review
+
+We believe `catch` to be a poor choice of keyword, because it:
+
+1. is used nowhere in other languages to demarcate the body which can result in
```

Comment:
As @nox mentioned on IRC, "nowhere in other languages" is incorrect; [Erlang uses it](http://learnyousomeerlang.com/errors-and-exceptions) in a similar way (See the "Wait, there's more!" section). Erlang _also_ has try..catch. So it's weird.

Reply:
Yes; a good catch - I'll update accordingly 👍